sklearn/_loss/_loss.pyx.tp

```diff
@@ -708,34 +707,47 @@ cdef inline double cgradient_half_binomial(
     #     return expit(raw_prediction) - y_true
     # i.e. no "if else" and an own inline implementation of expit instead of
     #     from scipy.special.cython_special cimport expit
-    # The case distinction raw_prediction < 0 in the stable implementation
-    # does not provide significant better precision. Therefore we go without
-    # it.
+    # The case distinction raw_prediction < 0 in the stable implementation does not
+    # provide significant better precision apart from protecting overflow of exp(..).
+    # The branch (if else), however, can incur runtime costs of up to 30%.
+    # Instead, we help branch prediction by almost always ending in the first if clause
+    # and making the second branch (else) a bit simpler. This has the exact same
+    # precision but is faster than the stable implementation.
+    # As branching criteria, we use the same cutoff as in log1pexp. Note that the
+    # maximal value to get gradient = -1 with y_true = 1 is -37.439198610162731
+    # (based on mpmath), and scipy.special.logit(np.finfo(float).eps) ~ -36.04365.
```

Comment:
Thank you for this comprehensive comment.

```suggestion
    # (based on the evaluation of mpmath's log1pexp for this case),
    # and scipy.special.logit(np.finfo(float).eps) ~ -36.04365.
```

Reply:
No, it's really the evaluation of the loss, gradient, hessian with the simplest formula, just with mpmath.dps >= 50.